src/main/java/com/google/sps/servlets/MealServlet.java

```diff
@@ -162,20 +162,24 @@ private void getMealById(HttpServletRequest request, HttpServletResponse respons
     private void returnIdOfSimilar(HttpServletRequest request, HttpServletResponse response) throws IOException {
         // Fuction redirect to random page
         // TODO(grenlayk): implement suggestions algorithm here for Product Alpha
+        Long mealId = Long.parseLong(getParameter(request, "id", "0"));
+
         Query query = new Query("Meal");
         DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
         PreparedQuery results = datastore.prepare(query);
         ArrayList<Long> idList = new ArrayList<>();
         for (Entity entity : results.asIterable()) {
             Long id = (Long)entity.getProperty("id");
-            idList.add(id);
+            if (id != null && !id.equals(mealId)) {
+                idList.add(id);
+            }
         }
```

Comment:
Nit:
I believe this block could be rewritten using the [Stream](https://docs.oracle.com/javase/8/docs/api/java/util/stream/Stream.html) module.

This looks like a combination on `map` and `filter`.

Not that it matters much, though.